tutorials/dataframe/df107_SingleTopAnalysis.py

```diff
@@ -0,0 +1,245 @@
+# Create a ROOT dataframe for each dataset
+# Note that we load the filenames from an external json file.
+path = "root://eospublic.cern.ch//eos/opendata/atlas/OutreachDatasets/2020-01-22"
+files = json.load(open(os.path.join(os.environ["ROOTSYS"], "tutorials/dataframe", "df107_SingleTopAnalysis.json")))
+processes = files.keys()
+df = {}
+xsecs = {}
+sumws = {}
+samples = []
+for p in processes:
+    for d in files[p]:
+        # Construct the dataframes
+        folder = d[0] # Folder name
+        sample = d[1] # Sample name
+        xsecs[sample] = d[2] # Cross-section
+        sumws[sample] = d[3] # Sum of weights
+        num_events = d[4] # Number of events
+        samples.append(sample)
+        df[sample] = ROOT.RDataFrame("mini", "{}/1lep/{}/{}.1lep.root".format(path, folder, sample))
+
+        # Scale down the datasets if requested
+        if not (args.t or lumi_scale == 1.0):
+            df[sample] = df[sample].Range(int(num_events * lumi_scale))
+
+# Select events for the analysis and make histograms of the top mass
+ROOT.gInterpreter.Declare("""
+using VecF_t = const ROOT::RVec<float>&;
+using VecI_t = const ROOT::RVec<int>&;
+int FindGoodLepton(VecI_t goodlep, VecI_t type, VecF_t pt, VecF_t eta, VecF_t phi, VecF_t e, VecF_t trackd0pv, VecF_t tracksigd0pv, VecF_t z0)
+{
+    int idx = -1;
+    for(auto i = 0; i < type.size(); i++) {
+        if(!goodlep[i]) continue;
+        if (type[i] == 11 && abs(eta[i]) < 2.47 && (abs(eta[i]) < 1.37 || abs(eta[i]) > 1.52) && abs(trackd0pv[i] / tracksigd0pv[i]) < 5) {
+            const ROOT::Math::PtEtaPhiEVector p(pt[i] / 1000.0, eta[i], phi[i], e[i] / 1000.0);
+            if (abs(z0[i] * sin(p.Theta())) < 0.5) {
+                if (idx == -1) idx = i;
+                else return -1;
+            }
+        }
+        if (type[i] == 13 && abs(eta[i]) < 2.5 && abs(trackd0pv[i] / tracksigd0pv[i]) < 3) {
+            const ROOT::Math::PtEtaPhiEVector p(pt[i] / 1000.0, eta[i], phi[i], e[i] / 1000.0);
+            if (abs(z0[i] * sin(p.Theta())) < 0.5) {
+                if (idx == -1) idx = i;
+                else return -1;
+            }
+        }
+    }
+    return idx;
+}
+
+float ComputeTransverseWMass(float lep_pt, float lep_eta, float lep_phi, float lep_e, float met_et, float met_phi)
+{
+    const ROOT::Math::PtEtaPhiEVector lep(lep_pt, lep_eta, lep_phi, lep_e);
+    const ROOT::Math::PtEtaPhiEVector met(met_et, 0, met_phi, met_et);
+    return sqrt(2 * lep.Pt() * met.Et() * (1 - cos(ROOT::Math::VectorUtil::DeltaPhi(lep, met))));
+}
+""")
+
+
+for s in samples:
+    # Select events with electron or muon trigger and with a missing tranverse energy above 30 GeV
+    df[s] = df[s].Filter("trigE || trigM")\
+                 .Filter("met_et > 30000")
+
+    # Perform preselection of highly isolated leptons
+    df[s] = df[s].Define("goodlep", "lep_isTightID && lep_pt > 35000 && lep_ptcone30 / lep_pt < 0.1 && lep_etcone20 / lep_pt < 0.1")\
+                 .Filter("Sum(goodlep) > 0")
```

Comment:
It would maybe be better to write explicitly `ROOT::VecOps::Sum`?

Reply:
I always wonder what is better. `ROOT::VecOps::Sum` is rather bulky, but it's for sure more clear what's going on!

Reply:
I think I agree that adding the namespace is better :thinking:

Reply:
Yes, it's better. What if somebody wants to see what's going on, and is looking for the docs?

Reply:
Yes I prefer the tutorials to be bulkier and clearer to help the users understand better :smile:  I always imagine myself trying to find the documentation for every single function call and writing the full namespace would spare some time :+1: